docs/source/notes/cuda.rst

```diff
@@ -54,6 +54,16 @@ Below you can find a small example showcasing this::
         f = torch.randn(2).cuda(cuda2)
         # d.device, e.device, and f.device are all device(type='cuda', index=2)
 
+.. note::
+
+    Starting in CUDA 12.0, the behavior of `cudaSetDevice`_ slightly changed resulting
+    in immediate context creation on the device. Subsequently, a call to
+    ``torch.cuda.set_device(d)`` will immediately create the CUDA context on the
+    device ``d`` if it was not created previously. This eager behavior could potentially
+    lead to GPU memory pollution. In PyTorch 2.1.0a0+git69eef5a and newer this issue was
```

Comment:
But we are still creating context with `torch.cuda.set_device(d)`? 
or are we creating contexts only with guards? Guard behavior change also has to mentioned. Previously in (python) guard context managers, if there were no cuda operations the context was not created, and it is created now. This is an edge case, but it's better to describe it.